src/libcore/internal_macros.rs

```diff
@@ -38,6 +38,7 @@ macro_rules! forward_ref_binop {
     };
     (impl $imp:ident, $method:ident for $t:ty, $u:ty, #[$attr:meta]) => {
         #[$attr]
+        #[must_use]
```

Comment:
Is the `impl` the right place for this attribute?  I thought the lint only looked at methods on trait definitions.

Reply:
Sorry, I didn't read the mentoring instructions before, but this fix needs tweaking:
- Instead of adding `#[must_use]` to method definitions in impls, you need to add them to the trait definitions (e.g. here: https://github.com/rust-lang/rust/blob/699eb95ae3dae023cddca85facf6660b196bd0d1/src/libcore/ops/arith.rs#L97).
- You don't need any of the `#[must_use]` in the macros, for the same reason.

Reply:
I don't fully understand what is going on in the `impl`, though I'm a little unsure of where else I would put the attribute.  The recurrence of `#[$attr]` in each `impl` made it seem as though the attributes need to be repeated before each possible combination of `T` and `U` states.

Reply:
Oh, ok.  I can go fix that.  Thanks for the quick support.  I also just learned that even though GitHub will automatically show new messages as they come in, I need to refresh to see edits.

Reply:
Take a look at `PartialOrd`, which implements `lt` (`<`): https://github.com/rust-lang/rust/blob/699eb95ae3dae023cddca85facf6660b196bd0d1/src/libcore/cmp.rs#L642-L650
The `#[must_use]` attribute is on the definition in `trait PartialOrd`, but not on any of the `impl PartialOrd`. The way `#[must_use]` works at the moment, it's only the trait declaration which is important — everything else inherits from the trait. (That way, _anyone_ who implements an operator will get the advantage of `#[must_use]` without having to add it themselves.